src/contentScript.ts

```diff
@@ -15,114 +15,79 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-import { v4 as uuidv4 } from "uuid";
-
-const PIXIEBRIX_SYMBOL = Symbol.for("pixiebrix-content-script");
-
-declare global {
-  interface Window {
-    [PIXIEBRIX_SYMBOL]?: string;
-  }
-}
-
-// Usages below are safe because we're using our own symbol defined above
-// eslint-disable-next-line security/detect-object-injection
-if (!window[PIXIEBRIX_SYMBOL]) {
-  // eslint-disable-next-line security/detect-object-injection
-  window[PIXIEBRIX_SYMBOL] = uuidv4();
-} else {
-  // Don't load script multiple times. Required since we can't do conditional imports to no-op the
-  // content script if it's already installed
-
-  console.debug(
-    // eslint-disable-next-line security/detect-object-injection -- safe because it's using the symbol we create
-    `PixieBrix contentScript already installed: ${window[PIXIEBRIX_SYMBOL]}`
-  );
-  throw Error(`PixieBrix contentScript already installed`);
-}
+const start = Date.now();
 
+import { v4 as uuidv4 } from "uuid";
 import "@/extensionContext";
-import { reportError } from "@/telemetry/logging";
-import { showConnectionLost } from "@/contentScript/connection";
-
-window.addEventListener("error", function (e) {
-  if (isConnectionError(e)) {
-    showConnectionLost();
-  } else {
-    reportError(e);
-    return false;
-  }
-});
-
-window.addEventListener("unhandledrejection", function (e) {
-  if (isConnectionError(e)) {
-    showConnectionLost();
-  } else {
-    reportError(e);
-  }
-});
-
+import addErrorListeners from "@/contentScript/errors";
 import "@/blocks";
 import "@/contrib";
 import "@/contentScript/devTools";
 import "@/contentScript/contextMenus";
 import "@/contentScript/browserAction";
-
-// Import for the side effect of registering js defined blocks
+import addContentScriptListener from "@/contentScript/backgroundProtocol";
 import { handleNavigate } from "@/contentScript/lifecycle";
-import "@/contentScript/externalProtocol";
-import { notifyReady, whoAmI } from "@/contentScript/executor";
+import addExternalListener from "@/contentScript/externalProtocol";
+import addExecutorListener, {
+  notifyReady,
+  whoAmI,
+} from "@/contentScript/executor";
 import "@/messaging/external";
 import "@/contentScript/script";
 import "@/vendors/notify";
 import { markReady, updateTabInfo } from "@/contentScript/context";
 import { initTelemetry } from "@/telemetry/events";
 import "@/contentScript/uipath";
-import { isConnectionError } from "@/errors";
 
-const start = Date.now();
+const PIXIEBRIX_SYMBOL = Symbol.for("pixiebrix-content-script");
+const uuid = uuidv4();
+
+declare global {
+  interface Window {
+    [PIXIEBRIX_SYMBOL]?: string;
+  }
+}
 
 async function init(): Promise<void> {
+  // Add error listeners first so they can catch any initialization errors
+  addErrorListeners();
```

Comment:
Nit: I'd pad it out to clarify that the comment is about this single line

```suggestion
  addErrorListeners();

```